flink-ml-core/src/test/java/org/apache/flink/ml/api/IntegrateWithServableTest.java

```diff
@@ -0,0 +1,106 @@
+import org.apache.flink.ml.servable.types.DataTypes;
+import org.apache.flink.ml.util.Serializer;
+import org.apache.flink.ml.util.TestUtils;
+import org.apache.flink.streaming.api.environment.StreamExecutionEnvironment;
+import org.apache.flink.table.api.Table;
+import org.apache.flink.table.api.bridge.java.StreamTableEnvironment;
+import org.apache.flink.test.util.AbstractTestBase;
+
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+import java.io.ByteArrayInputStream;
+import java.util.Arrays;
+import java.util.Collections;
+
+import static org.apache.flink.ml.servable.TestUtils.compareDataFrame;
+import static org.apache.flink.ml.util.TestUtils.saveAndLoadServable;
+
+/** Tests the behavior of integration with Servables. */
+public class IntegrateWithServableTest extends AbstractTestBase {
+
+    private StreamTableEnvironment tEnv;
+
+    private static final DataFrame INPUT =
+            new DataFrame(
+                    Collections.singletonList("input"),
+                    Collections.singletonList(DataTypes.INT),
+                    Arrays.asList(
+                            new Row(Collections.singletonList(1)),
+                            new Row(Collections.singletonList(2)),
+                            new Row(Collections.singletonList(3))));
+
+    private static final DataFrame EXPECTED_OUTPUT =
+            new DataFrame(
+                    Collections.singletonList("input"),
+                    Collections.singletonList(DataTypes.INT),
+                    Arrays.asList(
+                            new Row(Collections.singletonList(11)),
+                            new Row(Collections.singletonList(12)),
+                            new Row(Collections.singletonList(13))));
+
+    @Rule public final TemporaryFolder tempFolder = new TemporaryFolder();
+
+    @Before
+    public void before() {
+        StreamExecutionEnvironment env = TestUtils.getExecutionEnvironment();
+        tEnv = StreamTableEnvironment.create(env);
+    }
+
+    @Test
+    public void testSaveModelAndLoadAsServable() throws Exception {
+        SumModel model = new SumModel().setModelData(tEnv.fromValues(10));
+
+        SumModelServable servable =
+                saveAndLoadServable(tEnv, model, tempFolder.newFolder().getAbsolutePath());
+
+        DataFrame output = servable.transform(INPUT);
+
+        compareDataFrame(EXPECTED_OUTPUT, output);
+    }
+
+    @Test
+    public void testGetModelDataAndSetToServable() throws Exception {
+        SumModel model = new SumModel().setModelData(tEnv.fromValues(10));
+        Table modelDataTable = model.getModelData()[0];
+        Integer modelData =
+                tEnv.toDataStream(modelDataTable)
+                        .map(x -> (int) x.getField(0))
+                        .executeAndCollect()
+                        .next();
+
+        byte[] serializedModelData = Serializer.serialize(modelData);
```

Comment:
Serializing the model data as object stream is not very efficient, similar to the serialization using kryo.  And we don't expect user code to be directly aware of the internal data structure of the model data.

It would be useful for tests to show the best practice of developing and using servable.

How is one possible way to convert the a Table from `Model#getModelData` to an InputStream that can be provided to `ModelServable#setModelData`.

1. Table of model data object (in this case model object is an integer)
2. Table of byte[]
3. List of bytes[]
4. InputStream of bytes[]